### PR TITLE
feat: add --share-envs and --stream-output options

### PR DESCRIPTION
**Pull request by saartochner-lumigo**

### CODE REVIEW

File: src/bin/options/share-envs.ts
Comment (saartochner) [inline src/bin/options/share-envs.ts]: Note that we do the `require` only in the case that the user asked for this action.
We do so because `worker_threads` does not always exists

Comment (1999) [reply src/bin/options/share-envs.ts]: IIRC `worker_threads` doesn't exist for older Node.JS versions. So what will happen if the user specifies `--share-env` in an older Node.JS version? Do you think it's worth it to exit(1) with an error saying this is not supported?

**Review by 1999 — CHANGES_REQUESTED** [review]
Probably not strictly related to your changes but more to your use case: environment variables will be shared between the workers and the main process but how're you going to let worker B know about the changes in environment variables in worker A?

**Comment by saartochner** [comment]
> Probably not strictly related to your changes but more to your use case: environment variables will be shared between the workers and the main process but how're you going to let worker B know about the changes in environment variables in worker A?

Great question!
I wrote a beforeEach and afterEach hooks that capture and release the locks for the different services using the environment variables.

File: src/bin/options/share-envs.ts
Comment (saartochner) [reply src/bin/options/share-envs.ts]: I'll do it 👍

Comment (saartochner) [reply src/bin/options/share-envs.ts]: I'll do it 👍

**Comment by saartochner** [comment]
Typescript is failing because of this:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/41562
:(

**Review by 1999 — CHANGES_REQUESTED** [review]
@saartochner

> I wrote a beforeEach and afterEach hooks that capture and release the locks for the different services using the environment variables.

Could you please add something like a quick guide into README for those who would want to use this option?

**Comment by 1999** [comment]
Also, let's wait until your PR to @types/node gets merged and update the code in this PR.

**Comment by saartochner** [comment]
> Also, let's wait until your PR to @types/node gets merged and update the code in this PR.

Update: My PR is duplicated with: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/41356
so we just wait until this one will be approved 😩

File: package.json
Comment (saartochner) [inline package.json]: https://stackoverflow.com/questions/50616253/how-to-resolve-chicken-egg-situation-with-tsc-and-npm-install

**Pull request closed without merging**
